src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2912,6 +2925,14 @@ public int getCurrentGenerationNumber()
         return Gossiper.instance.getCurrentGenerationNumber(FBUtilities.getBroadcastAddress());
     }
 
+    public Instant getLastSuccessfulCleanupTsForNode()
+    {
+        if (cleanupState == null)
+            throw new IllegalStateException("Tracker is null, cleanup state info unretrievable.");
```

Comment:
If we fail to create the cleanup state tracker will all the cleanup endpoints that require it fail too until we restart c* and try again? If the initial creation fails, could we retry creating it on these endpoints to avoid the restart?

Reply:
So try again and then throw an error if it does not work on retry?